### ISSUE TITLE
Curl is missing from command

### ISSUE THREAD

**Issue by kroomagnon** [matteocorti/check_ssl_cert#76]
Hi Matteo,

I found that if I run `check_ssl_cert` only with the `--host` option, then `curl` is missing from the command:

```
> check_ssl_cert_1.65.0 -d --host netlock.hu
[DBG] Checking revokation via OCSP
[DBG] OCSP: fetching issuer certificate http://aia1.netlock.hu/index.cgi?ca=cbca to /tmp/check_ssl_cert_1.65.0W8PjcG
[DBG] executing with timeout (15s):  --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /tmp/check_ssl_cert_1.65.0W8PjcG
[DBG]   /usr/bin/timeout 15 /bin/sh -c " --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /tmp/check_ssl_cert_1.65.0W8PjcG"
[DBG] OCSP: issuer certificate type:  empty
```

But there is no problem running like this:

```
> check_ssl_cert_1.65.0 -d --host netlock.hu --check-ssl-labs A
[DBG] Checking revokation via OCSP
[DBG] OCSP: fetching issuer certificate http://aia1.netlock.hu/index.cgi?ca=cbca to /tmp/check_ssl_cert_1.65.0EhK8Ws
[DBG] executing with timeout (15s): /usr/bin/curl --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /tmp/check_ssl_cert_1.65.0EhK8Ws
[DBG]   /usr/bin/timeout 15 /bin/sh -c "/usr/bin/curl --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /tmp/check_ssl_cert_1.65.0EhK8Ws"
[DBG] OCSP: issuer certificate type:  PEM certificate
```

**Comment by matteocorti** [comment]
Hi, I cannot reproduce the problem:

```
$ ./check_ssl_cert -d --host netlock.hu
[DBG] ROOT_CA = 
expect available (/usr/bin/expect)
timeout available (/opt/local/bin/timeout)
[DBG] perl available: /Users/corti/perl5/perlbrew/perls/perl-5.26.0/bin/perl
[DBG] date available: /bin/date
found BSD date with timestamp support: enabling date computations
[DBG] check_ssl_cert version: 1.65.0
[DBG] OpenSSL binary: /opt/local/bin/openssl
[DBG] OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
[DBG] OpenSSL configuration directory: /opt/local/etc/openssl
[DBG] 133 root certificates installed by default
[DBG] System info: Darwin id-als-docking-053.ethz.ch 17.5.0 Darwin Kernel Version 17.5.0: Mon Mar  5 22:24:32 PST 2018; root:xnu-4570.51.1~1/RELEASE_X86_64 x86_64
[DBG] Date computation: BSD
[DBG] '/opt/local/bin/openssl s_client' supports '-servername': using -servername netlock.hu
'/opt/local/bin/openssl s_client' does not support '-xmpphost': disabling 'to' attribute
downloading certificate to /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/
[DBG] netlock.hu is not an IP address
[DBG] executing with timeout (15s): echo 'Q' | /opt/local/bin/openssl s_client   -connect netlock.hu:443 -servername netlock.hu -verify 6     2> /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.IgakZg7n 1> /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.6QeSoTIZ
[DBG]   /opt/local/bin/timeout 15 /bin/sh -c "echo 'Q' | /opt/local/bin/openssl s_client   -connect netlock.hu:443 -servername netlock.hu -verify 6     2> /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.IgakZg7n 1> /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.6QeSoTIZ"
[DBG] storing a copy of the retrieved certificate in netlock.hu.crt
[DBG] storing a copy of the OpenSSL errors in netlock.hu.error
parsing the x509 certificate file
[DBG] subject= C = HU, L = Budapest, O = NETLOCK Kft., CN = www.netlock.hu
[DBG] CN         = www.netlock.hu
[DBG] CA_O       = NetLock Kft.
[DBG] CA_CN      = NetLock Üzleti (Class B) Tanúsítványkiadó
[DBG] SERIAL     = 49DD0FE61809F814EA44ED1C6E4D
[DBG] FINGERPRINT= 38:6D:6B:8F:47:DD:6C:AD:EE:07:D8:49:FE:F8:65:CB:23:00:7D:18
[DBG] OCSP_URI   = http://ocsp1.netlock.hu/cbca.cgi
[DBG] ISSUER_URI = http://aia1.netlock.hu/index.cgi?ca=cbca
[DBG]     Signature Algorithm: sha256WithRSAEncryption
[DBG] Date computations: BSD
The certificate will expire in 0 day(s)
[DBG] subjectAlternativeName = www.netlock.hu www.netlock.net netlock.hu netlock.net *.netlock.hu *.netlock.net minositett.netlock.hu minositett.netlock.net
[DBG] Checking expiration date
[DBG] Checking revokation via OCSP
[DBG] OCSP: fetching issuer certificate http://aia1.netlock.hu/index.cgi?ca=cbca to /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.fyP4ys9G
[DBG] executing with timeout (15s): /opt/local/bin/curl --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.fyP4ys9G
[DBG]   /opt/local/bin/timeout 15 /bin/sh -c "/opt/local/bin/curl --silent --location http://aia1.netlock.hu/index.cgi?ca=cbca > /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.fyP4ys9G"
[DBG] OCSP: issuer certificate type:  PEM certificate
[DBG] OCSP: storing a copy of the retrieved issuer certificate to index.cgi?ca=cbca
[DBG] OCSP: host = ocsp1.netlock.hu
[DBG] openssl ocsp supports the -header option
[DBG] openssl ocsp -header requires 'key value'
[DBG] executing /opt/local/bin/openssl ocsp -no_nonce -issuer /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.fyP4ys9G -cert /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.6QeSoTIZ  -url http://ocsp1.netlock.hu/cbca.cgi  -header HOST ocsp1.netlock.hu
[DBG] OCSP: response = Response Verify Failure
[DBG] OCSP: response = 140735726597064:error:27069065:OCSP routines:OCSP_basic_verify:certificate verify error:ocsp_vfy.c:140:Verify error:unable to get local issuer certificate
[DBG] OCSP: response = /var/folders/sq/sb7jrfk14qv7yyh9z55mszp40000gn/T/check_ssl_certXXXXXX.6QeSoTIZ: good
[DBG] OCSP: response = 	This Update: Apr  5 10:58:53 2018 GMT
SSL_CERT OK - x509 certificate 'www.netlock.hu' from 'NetLock Üzleti (Class B) Tanúsítványkiadó' valid until Apr  6 08:29:01 2018 GMT (expires today)|days=0;;;;
```

Could you please try to re-download the package (after releasing a buggy version was online for a short time). If it still does not work, could you please post the full debugging output?

**Comment by walley** [comment]
same problem here, I have curl in /usr/bin/curl but I have to add --curl-bin /usr/bin/curl, othervise it just executes 

`[DBG] executing with timeout (15s):  --silent --location http://crt.comodoca.com/COMODORSADomainValidationSecureServerCA.crt > /tmp/`

whole command is:
`/usr/local/nagios/libexec/check_ssl_cert --openssl /usr/local/bin/openssl -s -r /etc/pki/tls/certs/ca-bundle.crt -H dokumenty.hasici-ol.cz -p 443 --curl-bin /usr/bin/curl -d`

full debug:

[DBG] OpenSSL configuration directory: 
[DBG] 0 root certificates installed by default
[DBG] System info: Linux nagios.hasici-ol.cz 2.6.24.2 #7 SMP Tue Feb 26 12:12:17 CET 2008 i686 i686 i386 GNU/Linux
[DBG] Date computation: GNU
[DBG] '/usr/local/bin/openssl s_client' supports '-servername': using -servername dokumenty.hasici-ol.cz
[DBG] '/usr/local/bin/openssl s_client' supports '-xmpphost': using -xmpphost dokumenty.hasici-ol.cz
downloading certificate to /tmp
[DBG] dokumenty.hasici-ol.cz is not an IP address
[DBG] executing with timeout (15s): echo 'Q' | /usr/local/bin/openssl s_client   -connect dokumenty.hasici-ol.cz:443 -servername dokumenty.hasici-ol.cz -verify 6 -CAfile /etc/pki/tls/certs/ca-bundle.crt    2> /tmp/check_ssl_certR19889 1> /tmp/check_ssl_certg19887
[DBG] storing a copy of the retrieved certificate in dokumenty.hasici-ol.cz.crt
[DBG] storing a copy of the OpenSSL errors in dokumenty.hasici-ol.cz.error
parsing the x509 certificate file
[DBG] subject=OU = Domain Control Validated, OU = PositiveSSL Wildcard, CN = *.hasici-ol.cz
[DBG] CN         = *.hasici-ol.cz
[DBG] CA_O       = COMODO CA Limited
[DBG] CA_CN      = COMODO RSA Domain Validation Secure Server CA
[DBG] SERIAL     = F3456C506860BC86E8B0345EF1041249
[DBG] FINGERPRINT= 19:7C:02:D8:CD:5E:5D:76:B9:4B:38:84:1A:68:C3:01:C2:4B:6D:A5
[DBG] OCSP_URI   = http://ocsp.comodoca.com
[DBG] ISSUER_URI = http://crt.comodoca.com/COMODORSADomainValidationSecureServerCA.crt
[DBG]         Signature Algorithm: sha256WithRSAEncryption
[DBG] Date computations: GNU
The certificate will expire in 756 day(s)
[DBG] subjectAlternativeName = *.hasici-ol.cz hasici-ol.cz
[DBG] Checking expiration date
[DBG] Checking revokation via OCSP
[DBG] OCSP: fetching issuer certificate http://crt.comodoca.com/COMODORSADomainValidationSecureServerCA.crt to /tmp/check_ssl_certO19891
[DBG] executing with timeout (15s):  --silent --location http://crt.comodoca.com/COMODORSADomainValidationSecureServerCA.crt > /tmp/check_ssl_certO19891
/bin/sh: --silent: command not found
[DBG] OCSP: issuer certificate type:  empty
SSL_CERT UNKNOWN dokumenty.hasici-ol.cz: Unable to fetch OCSP issuer certificate.

**Comment by matteocorti** [comment]
Was fixed in 1.66.0 (forgot to close)
